feedback-module/src/components/pages/Screen2.js

```diff
@@ -8,34 +8,66 @@ import ModuleButton from "../common/Button";
 
 function Screen2({ feedbackType }) {
   const [checkedFields, setCheckedFields] = useState({});
+  const [otherField, setOtherField] = useState("");
+  const [inputQuestions, setInputQuestions] = useState({});
 
   useEffect(() => {
     let fields = {};
+
     feedbackType.checkboxes.map((label) => {
       fields = {
         ...fields,
         [label]: false,
       };
     });
     setCheckedFields(fields);
+    let questions = {};
+    feedbackType.textInputs.map((question) => {
+      questions = {
+        ...questions,
+        [question]: "",
+      };
+    });
+    setInputQuestions(questions);
   }, []);
 
   const onCheck = (label) => {
     let checked = checkedFields;
     checked[label] = !checked[label];
     setCheckedFields(checked);
-    console.log("hi");
+  };
+
+  const onSubmit = (e) => {
+    let checked = checkedFields;
+    checkedFields["Other"] &&
+      (checked = {
+        ...checked,
+        OtherText: otherField,
+      });
+    console.log(checked);
+    setCheckedFields(checked);
+    console.log(inputQuestions);
+    e.preventDefault();
   };
 
   return (
     <Module>
       <h1>{feedbackType.title}</h1>
       <Form className="maxw-none overflow-hidden">
-        <CheckboxList feedbackType={feedbackType} onCheck={() => onCheck} />
-        <TextboxList feedbackType={feedbackType} />
+        <CheckboxList
+          feedbackType={feedbackType}
```

Comment:
Minor point, but would it be possible to just directly pass in feedbackType.checkboxes? It doesn't seem like you're using any other part of feedbackType within the CheckboxList component

Reply:
Yes definitely good point, i'll fix that now

Reply:
I also changed it for the textboxList as well